src/main/scala/com/datastax/sparkstress/SparkCassandraStress.scala

```diff
@@ -157,61 +162,74 @@ object SparkCassandraStress {
         .set("spark.cassandra.connection.keep_alive_ms", (config.streamingBatchIntervalSeconds*1000*5).toString)
         .setAll(config.sparkOps)
 
-    val sc = ConnectHelper.getContext(sparkConf)
+    val ss = ConnectHelper.getSparkSession(sparkConf)
     
     if (config.verboseOutput) { 
       println("\nDumping debugging output")
-      println(sc.getConf.toDebugString+"\n")
+      println(ss.sparkContext.getConf.toDebugString+"\n")
     }
 
     val test: StressTask =
       config.testName.toLowerCase match {
           /** Write Tasks **/
-        case "writeshortrow" => new WriteShortRow(config, sc)
-        case "writewiderow" => new WriteWideRow(config, sc)
-        case "writeperfrow" => new WritePerfRow(config, sc)
-        case "writerandomwiderow" => new WriteRandomWideRow(config, sc)
-        case "writewiderowbypartition" => new WriteWideRowByPartition(config, sc)
-
-          /** Read Tasks **/
-        case "pdcount" => new PDCount(config, sc)
-        case "ftsallcolumns" => new FTSAllColumns(config, sc)
-        case "ftsfivecolumns" => new FTSFiveColumns(config, sc)
-        case "ftsonecolumn" => new FTSOneColumn(config, sc)
-        case "ftspdclusteringallcolumns" => new FTSPDClusteringAllColumns(config, sc)
-        case "ftspdclusteringfivecolumns" => new FTSPDClusteringFiveColumns(config, sc)
-        case "jwcallcolumns" => new JWCAllColumns(config, sc)
-        case "jwcpdclusteringallcolumns" => new JWCPDClusteringAllColumns(config, sc)
-        case "jwcrpallcolumns" => new JWCRPAllColumns(config, sc)
-        case "retrievesinglepartition" => new RetrieveSinglePartition(config, sc)
-
-          /** Streaming Tasks **/
-        case "streamingwrite" => new StreamingWrite(config, sc)
+        case "writeshortrow" => new WriteShortRow(config, ss)
```

Comment:
maybe it would be easier to pickup all Task subclasses from the classpath and reflectively create an instance here?

Reply:
I like this idea